api/server/v1/marshaler.go

```diff
@@ -643,6 +643,106 @@ func (x *SubscribeResponse) MarshalJSON() ([]byte, error) {
 	return json.Marshal(resp)
 }
 
+func (x *InsertUserMetadataRequest) UnmarshalJSON(data []byte) error {
+	var mp map[string]jsoniter.RawMessage
+	if err := jsoniter.Unmarshal(data, &mp); err != nil {
+		return err
+	}
+	for key, value := range mp {
+		switch key {
+		case "metadataKey":
+			if err := jsoniter.Unmarshal(value, &x.MetadataKey); err != nil {
+				return err
+			}
+		case "value":
+			var valMp map[string]jsoniter.RawMessage
```

Comment:
should this be just:
x.Value = &UserMetadataValue{
  Value: value,
}

Reply:
synced up offline and made value to be freeform.